src/deadline/client/cli/_groups/queue_group.py

```
@@ -213,6 +212,10 @@ def queue_get(**args):
 @cli_queue.command(name="incremental-output-download")
 @click.option("--farm-id", help="The AWS Deadline Cloud Farm to use.")
 @click.option("--queue-id", help="The AWS Deadline Cloud Queue to use.")
+@click.option(
```

Can we add the usual reg-ex checks for the storage profile ID? Its too easy to type in the wrong ID and then getting confused with some call stack.

---

I looked at the other place with a `--storage-profile-id` option, in the job bundle submission, but didn't find code matching what you described. Do you have more details?

---

I mean apply a general `sp-{uuid}` input validation generally so customers get early notice of a failure. Instead of getting a backend error message mid way through the code execution.

---

I don't think adding a regex here is a good idea, but calling get_storage_profile_for_queue at the start of the command to fail early on start for malformed ids and other issues sounds nice. Here's the error message the service returns for a malformed id:

```
$ aws deadline get-storage-profile-for-queue --farm-id `deadline config get defaults.farm_id` --queue-id `deadline config get defau
lts.queue_id` --storage-profile-id xyz --profile `deadline config get defaults.aws_profile_name`

An error occurred (ValidationException) when calling the GetStorageProfileForQueue operation: Total 1 validation error found
 storageProfileId
 string does not match regex "^sp-[0-9a-f]{32}$"
```

---

I've updated the PR to get the storage profile early and write a better message:

```
$ deadline queue incremental-output-download --force-bootstrap --dry-run --bootstrap-lookback-minutes 3000 --storage-profile-id sp-12345
Could not retrieve the storage profile 'sp-12345', provided with --storage-profile-id, from Deadline Cloud:
An error occurred (ValidationException) when calling the GetStorageProfileForQueue operation: Total 1 validation error found
 storageProfileId
 string does not match regex "^sp-[0-9a-f]{32}$"
```

---

Thanks! This would work just as well.